app/controllers/settings/registry_mirrors_controller.rb

```diff
@@ -0,0 +1,100 @@
+# Settings::RegistryMirrorsController is responsibe to manage all the requests
+# related to the registry mirrors feature
+class Settings::RegistryMirrorsController < SettingsController
+  before_action :set_registry_mirror, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @grouped_mirrors = Registry.grouped_mirrors
+  end
+
+  def new
+    @registry_mirror = RegistryMirror.new
+    @cert = Certificate.new
+  end
+
+  def create
+    @registry = Registry.find(registry_mirror_params[:registry_id])
+    registry_mirror_create_params = registry_mirror_params.except(:certificate, :registry_id)
+    @registry_mirror = @registry.registry_mirrors.build(registry_mirror_create_params)
+    @cert = Certificate.find_or_initialize_by(certificate: certificate_param)
+
+    ActiveRecord::Base.transaction do
+      begin
+        @registry_mirror.save!
+
+        create_or_update_certificate! if certificate_param.present?
+
+        @created = true
+      rescue StandardError
+        raise ActiveRecord::Rollback
+      end
+    end
+
+    if @created
+      redirect_to [:settings, @registry_mirror], notice: "Mirror was successfully created."
+    else
+      render action: :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+    @cert = @registry_mirror.certificate || Certificate.new
+  end
+
+  # rubocop:disable Metrics/PerceivedComplexity
+  def update
+    @cert = @registry_mirror.certificate || Certificate.new(certificate: certificate_param)
+
+    ActiveRecord::Base.transaction do
+      begin
+        registry_mirror_update_params = registry_mirror_params.except(:certificate, :registry_id)
+        @registry_mirror.update_attributes!(registry_mirror_update_params)
+
+        if certificate_param.present?
+          create_or_update_certificate!
+        elsif @registry_mirror.certificate.present?
+          @registry_mirror.certificate.destroy!
+        end
+
+        @updated = true
+      rescue StandardError
+        raise ActiveRecord::Rollback
```

Comment:
Same with this block, we can remove the inner block and just let exceptions be raised without any `rescue` here.